lib/apiservers/portlayer/restapi/handlers/storage_handlers.go

```diff
@@ -507,6 +515,142 @@ func (h *StorageHandlersImpl) VolumeJoin(params storage.VolumeJoinParams) middle
 	return storage.NewVolumeJoinOK().WithPayload(actualHandle.String())
 }
 
+func (h *StorageHandlersImpl) StatPath(params storage.StatPathParams) middleware.Responder {
+	// do offline container stat path, if fails do online
+	op := trace.NewOperation(context.Background(), fmt.Sprintf("StatPath(%s)", params.DeviceID))
+
+	fileStat, err := h.volumeCache.StatPath(op, params.DeviceID, params.TargetPath)
+	if err != nil {
+		//fileStat, err = h.imageCache.StatPath(op, params.DeviceID, params.TargetPath)
+		// for debugging only:
+		return storage.
+		NewStatPathOK().
+			WithMode(1).
+			WithLinkTarget("symlink").
+			WithName(err.Error()).
+			WithSize(60)
+	}
+
+	// offline succeeded
+	if err == nil {
+		return storage.
+		NewStatPathOK().
+			WithMode(fileStat.Mode).
+			WithLinkTarget(fileStat.LinkTarget).
+			WithName(fileStat.Name).
+			WithSize(fileStat.Size)
+	}
+
+	//TODO: if throws disk in use error, proceed to online
+
+	// assume it's online container and obj id is container id first.
+	defer trace.End(trace.Begin(params.DeviceID))
+
+	vc := epl.Containers.Container(params.DeviceID)
+	if vc == nil {
+		log.Debugln("%#v", spew.Sdump(epl.Containers))
```

Comment:
We should be using the `op trace.opertation` here. so `op.Debug()` spew may also be a pretty heavy log here. We might need to find some key information for us to log. maybe the container ID. and the mountspecs possibly.